dubbo-config/dubbo-config-spring/src/main/java/org/apache/dubbo/config/spring/context/DubboBootstrapApplicationListener.java

```diff
@@ -43,6 +43,10 @@
 
     private final DubboBootstrap dubboBootstrap;
 
+    public DubboBootstrapApplicationListener() {
+        this.dubboBootstrap = DubboBootstrap.getInstance();
```

Comment:
If init by this constructor, the applicationContext in super class `OnceApplicationContextEventListener` will null, when call getApplicationContext(), cause NPE, I'm not sure this case will be happen.

Reply:
hi @horizonzy ,

I am not quite sure which class you mentioned which has `getApplicationContext()`.

I have found there are two places use `applicationContext`

- OnceApplicationContextEventListener

`ApplicationContextAware` has been implemented by `OnceApplicationContextEventListener`, so `applicationContext` will be automaticly injected by `public final void setApplicationContext(ApplicationContext applicationContext) throws BeansException`

- DubboBootstrapStartStopListenerSpringAdapter

The methods in `DubboBootstrapStartStopListenerSpringAdapter` will be called after `DubboBootstrap` started and before this `ApplicationContextEvent` will notify `DubboLifecycleComponentApplicationListener` with `applicationContext`.